settings_page.ml

```diff
@@ -1,9 +1,87 @@
-let settings_layout (configuration : Configuration.t) =
+let config_table_row (configuration : Configuration.t) =
   let ip = Ipaddr.to_string configuration.server_ip in
   let port = string_of_int configuration.server_port in
   let certificate = X509.Certificate.encode_pem configuration.certificate in
   let private_key = X509.Private_key.encode_pem configuration.private_key in
   let last_update = Utils.TimeHelper.string_of_ptime configuration.updated_at in
+  Tyxml_html.(
+    tr
+      [
+        td
+          ~a:
+            [
+              a_class
+                [
+                  "px-6 py-4 whitespace-nowrap text-sm font-medium \
+                   text-gray-800";
+                ];
+            ]
+          [ txt configuration.name ];
+        td
+          ~a:
+            [
+              a_class
+                [
+                  "px-6 py-4 whitespace-nowrap text-sm font-medium \
+                   text-gray-800";
+                ];
+            ]
+          [
+            txt (String.sub certificate 10 30 ^ "...");
+            p [ txt (String.sub private_key 10 30 ^ "...") ];
```

Comment:
not sure this was there before, but is the `String.sub` safe here? and why use an offset of 10 and a length of 30? I'd appreciate a function that takes a string and returns a string where it potentially uses sub (if the incoming string is longer than N characters).